core/services/workflows/engine.go

```diff
@@ -92,7 +92,10 @@ func (e *Engine) resolveWorkflowCapabilities(ctx context.Context) error {
 		}
 	}
 	if !triggersInitialized {
-		return fmt.Errorf("failed to resolve triggers")
+		return &workflowError{reason: "failed to resolve triggers", labels: map[string]string{
+			wIDKey: e.workflow.id,
+		},
+		}
```

Comment:
Nit: this indentation is a bit odd -- gofmt won't correct these sorts of multiline indent issues normally